frontend/src/components/Login/signin.tsx

```diff
@@ -0,0 +1,103 @@
+import './Sign.css';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import { useNavigate } from 'react-router-dom';
+
+interface FormValues {
+  email: string;
+  password: string;
+}
+interface ApiResponse {
+  result: any;
+ // auth: string;
+}
+const Signin = () => {
+  const navigate = useNavigate();
+  const formik = useFormik<FormValues>({
+    initialValues: {
+      email: '',
+      password: '',
+    },
+   
+    validationSchema: Yup.object({
+      email: Yup.string().email('Invalid email address').required('Email is required'),
+      password: Yup.string()
+        .required('Password is required')
+        .min(8, 'Password must be at least 8 characters long'),
+    }),
+
+    onSubmit: (values, { resetForm }) => {
+      console.log(values);
+      fetch('http://localhost:5000/login', {
```

Comment:
use axios to make api calls and also please write proper services for making API calls